### ISSUE TITLE
API change

### ISSUE THREAD

**Issue by viking-sudo-rm** [viking-sudo-rm/stacknn-core#1]
Should change API from classmethod `empty()` and method `update()` to two methods `reset()`and `update()`.

This would make `get_num_actions()` a property of an instance, not a class. This makes it easier to change the number of actions for multipush/multipop stacks.

**Comment by viking-sudo-rm** [comment]
Should be able to do this in a way that doesn't break old API. In other words, we define

```python
@classmethod
def empty(cls, batch_size, stack_dim, device):
    stack = cls(stack_dim)
    stack.reset(batch_size, device)
    return stack
```

**Comment by viking-sudo-rm** [comment]
Fixed in 95446f2.
